### DISCUSSION TITLE
Can we have the code for exporting PDF files as in the DEMO?

### DISCUSSION THREAD

**Discussion by fablau** [opensheetmusicdisplay/opensheetmusicdisplay#1350] [Q&A]
Thank you for making OSMD, it's a great system indeed.

I am testing it extensively, and while I was exploring your demo page below:

https://opensheetmusicdisplay.github.io/demo/

I am unable to understand what's the process you use to export PDF files from the scores. I guess you are using svg2pdf or something else based on jsPDF, but it'd be great if you could show us an un-minified version of the JavaScript code behind your demo in order to easily see how every aspect of what's showcased there works.

In particularly, I am unable to understand how PDF exporting works.

Thank you!

**Comment by sschmidTU** [comment]
There are no secrets here, the code is in the demo :)
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/demo/index.js#L842

It's not in main OSMD because it would add to the OSMD library size, and you are encouraged to research and implement your own PDF export. The libraries we use actually have some issues and don't always export correctly, but we haven't found better ones yet. If you do, please let us know!

**Reply by fablau** [reply]
Thank you so very much! I completely missed that! I appreciate your help, and of course, I'll share anything I improve that could help the community here.

Thanks again.
